### ISSUE TITLE
[bug] Failed to build on linux/386 with latest tcp redir support

### ISSUE THREAD

**Issue by 1715173329** [HyNetwork/hysteria#357]
```
$ env CGO_ENABLED=0 GOOS=linux GOARCH=386 go build -ldflags='-s -w'
# github.com/tobyxdd/hysteria/pkg/redirect
../pkg/redirect/tcp_linux.go:93:41: undefined: syscall.SYS_GETSOCKOPT
../pkg/redirect/tcp_linux.go:97:41: undefined: syscall.SYS_GETSOCKOPT
```

CI threw the same error too:
https://github.com/HyNetwork/hysteria/runs/6767790534?check_suite_focus=true#step:4:182

**Comment by tobyxdd** [comment]
Fixed in 8a64099a9633f5537c41d4e12ea44870fe5d2e38

**Comment by 1715173329** [comment]
Thanks for the quick response!
